src/settings.cpp

```diff
@@ -2204,26 +2204,20 @@ void IConsoleSetSetting(const char *name, int value)
  */
 void IConsoleGetSetting(const char *name, bool force_newgame)
 {
-	char value[20];
 	const SettingDesc *sd = GetSettingFromName(name);
-	const void *ptr;
-
 	if (sd == nullptr) {
 		IConsolePrintF(CC_WARNING, "'%s' is an unknown setting.", name);
 		return;
 	}
 
-	ptr = GetVariableAddress((_game_mode == GM_MENU || force_newgame) ? &_settings_newgame : &_settings_game, &sd->save);
+	const void *object = (_game_mode == GM_MENU || force_newgame) ? &_settings_newgame : &_settings_game;
 
 	if (sd->cmd == SDT_STDSTRING) {
+		const void *ptr = GetVariableAddress(object, &sd->save);
 		IConsolePrintF(CC_WARNING, "Current value for '%s' is: '%s'", name, reinterpret_cast<const std::string *>(ptr)->c_str());
 	} else {
-		if (sd->cmd == SDT_BOOLX) {
-			seprintf(value, lastof(value), (*(const bool*)ptr != 0) ? "on" : "off");
-		} else {
-			seprintf(value, lastof(value), sd->min < 0 ? "%d" : "%u", (int32)ReadValue(ptr, sd->save.conv));
-		}
-
+		char value[20];
+		sd->FormatValue(value, lastof(value), object);
```

Comment:
shouldn't this be `ptr`? `object` is always `_settings_newgame` or `_settings_game` now,so the first object, not?

Reply:
Just as check in that case: a bit lower you do still use `ptr`. Is that correct, or should that be `GetGameSettings` instead?

Reply:
Ah, now I see what you meant. That `ptr` is indeed wrong.